compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -604,6 +609,13 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
             if RUSTC_SPECIFIC_FEATURES.contains(&feature) {
                 return None;
             }
+
+            // if the target-feature is "backchain" and LLVM version is greater than 18
+            // then we also need to add "+backchain" to the target-features attribute.
+            // otherwise, we will only add the naked `backchain` attribute to the attribute-group.
+            if feature == "backchain" && llvm_major < 18 {
```

Comment:
This entire `RUSTC_SPECIAL_FEATURES` is a temporary hack we can drop when we start requiring LLVM 18, right? Please file an issue to track this cleanup.

Reply:
> This entire `RUSTC_SPECIAL_FEATURES` is a temporary hack we can drop when we start requiring LLVM 18, right? Please file an issue to track this cleanup.

Yes, this is due to older LLVM does not support adding this attribute to the target-features attribute.